samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -262,3 +262,27 @@ def expect_minimum_argument_count!(arguments, min_count)
   false
 }
 
+ZIL_BUILTINS[:OBJECT] = lambda { |arguments, context|
+  # Objects have a name, and a list of properties and values
+
+  raise FunctionError, "OBJECT requires at least a name!" unless arguments.length > 0
+
+  # not sure if this is needed
+  raise FunctionError, "OBJECT requires properties" unless arguments.length > 1
+
+  object_name, *object_properties = arguments
+
+  object = { name: object_name, properties: {} }
+
+  object_properties.each do |property|
+    raise FunctionError, "Parameters to object must be in list form" unless property.class == Syntax::List
+    raise FunctionError, "OBJECT properties require a name and values!" unless property.elements.length > 1
+
+    property_name, *property_values = property.elements
+
+    object[:properties][property_name] = property_values.length == 1 ? property_values[0] : property_values
+  end
+
+  object
+}
```

Comment:
this should assign the object to the global variable under it's name, not return it, thus the creation of `object` is not needed as it currently stands, it should be a bare hash (but again I may be wrong (cc) @kfischer-okarin)